src/MongoDB.Driver.Core/Core/Configuration/CryptClientSettings.cs

```diff
@@ -0,0 +1,122 @@
+    /// Represents settings for a crypt client.
+    /// </summary>
+    public sealed class CryptClientSettings
+    {
+        /// <summary>
+        /// Gets a value indicating whether query analysis should be bypassed.
+        /// </summary>
+        public bool? BypassQueryAnalysis { get; }
+
+        /// <summary>
+        /// Gets the csfle library path.
+        /// </summary>
+        public string CsfleLibPath { get; }
+
+        /// <summary>
+        /// Gets the csfle search path.
+        /// </summary>
+        public string CsfleSearchPath { get; }
+
+        /// <summary>
+        /// Gets the encrypted fields map.
+        /// </summary>
+        public IReadOnlyDictionary<string, BsonDocument> EncryptedFieldsMap { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether csfle is required.
+        /// </summary>
+        public bool? IsCsfleRequired { get; }
+
+        /// <summary>
+        /// Gets the KMS providers.
+        /// </summary>
+        public IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> KmsProviders { get; }
+
+        /// <summary>
+        /// Gets the schema map.
+        /// </summary>
+        public IReadOnlyDictionary<string, BsonDocument> SchemaMap { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CryptClientSettings"/> class.
+        /// </summary>
+        public CryptClientSettings()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CryptClientSettings"/> class.
+        /// </summary>
+        /// <param name="bypassQueryAnalysis">The bypass query analysis.</param>
+        /// <param name="csfleLibPath">The csfle library path.</param>
+        /// <param name="csfleSearchPath">The csfle search path.</param>
+        /// <param name="encryptedFieldsMap">The encrypted fields map.</param>
+        /// <param name="isCsfleRequired">Value indicating whether csfle is required..</param>
+        /// <param name="kmsProviders">The KMS providers.</param>
+        /// <param name="schemaMap">The schema map.</param>
+        public CryptClientSettings(
+            bool? bypassQueryAnalysis,
+            string csfleLibPath,
+            string csfleSearchPath,
+            IReadOnlyDictionary<string, BsonDocument> encryptedFieldsMap,
+            bool? isCsfleRequired,
+            IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> kmsProviders,
+            IReadOnlyDictionary<string, BsonDocument> schemaMap)
+        {
+            BypassQueryAnalysis = bypassQueryAnalysis;
+            CsfleLibPath = csfleLibPath;
+            CsfleSearchPath = csfleSearchPath;
+            EncryptedFieldsMap = encryptedFieldsMap;
+            IsCsfleRequired = isCsfleRequired;
+            KmsProviders = kmsProviders;
+            SchemaMap = schemaMap;
+        }
+
+        // methods
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
```

Comment:
unnecessary?

Reply:
let's use `Hasher` implementation as we do in other places:

               return new Hasher()
                .HashElements(_credentials)
                .HashElements(_servers)
                .GetHashCode();

Reply:
No really need for, as we don't use this class in any hash based collection, and full `Equals` is always called.

Reply:
maybe but It confuses to override a method and then just call his base implementation.